include/lsst/afw/image/VisitInfo.h

```diff
@@ -0,0 +1,217 @@
+                ///< X is flipped, if necessary, to match the handedness of Az,Alt.
+    MOUNT       ///< The position sent to the instrument rotator; the details depend on the rotator.
+};
+
+/**
+ * Information about a single exposure of an imaging camera.
+ *
+ * Includes exposure duration and date, and telescope pointing and orientation.
+ *
+ * All information is for the middle of the exposure
+ * and at the boresight (center of the focal plane).
+ * Thus for a mosaic camera VisitInfo is the same for all detectors in the mosaic.
+ *
+ * VisitInfo is immutable.
+ */
+class VisitInfo : public table::io::PersistableFacade<VisitInfo>, public table::io::Persistable {
+public:
+
+    /**
+     * Construct a VisitInfo
+     *
+     * @param[in] exposureId  exposure ID
+     * @param[in] exposureTime  exposure duration (sec)
+     * @param[in] date  TAI (international atomic time) MJD date at middle of exposure
+     * @param[in] ut1  UT1 (universal time) MJD date at middle of exposure
+     * @param[in] era  earth rotation angle at middle of exposure
+     * @param[in] boresightRaDec  ICRS RA/Dec of boresight at middle of exposure
+     * @param[in] boresightAzAlt  refracted apparent topocentric Az/Alt of boresight at middle of exposure;
+                    for now this is represented as a plain coord::Coord,
+                    but in the long run it will be a SphPoint or a specialized Coord if one is added
+                    (we have TopocentricCoord, but not with refraction)
+     * @param[in] boresightAirmass  airmass at the boresight, relative to zenith at sea level
+     * @param[in] boresightRotAngle  rotation angle at boresight at middle of exposure;
+                        see getBoresightRotAngle for details
+     * @param[in] rotType  rotation type
+     * @param[in] observatory  observatory longitude, latitude and altitude
+     * @param[in] weather  basic weather information for computing air mass
+     */
+    explicit VisitInfo(
+        table::RecordId exposureId,
+        double exposureTime,
+        daf::base::DateTime const & date,
+        double ut1,
+        geom::Angle const & era,
+        coord::IcrsCoord const & boresightRaDec,
+        coord::Coord const & boresightAzAlt,
+        double boresightAirmass,
+        geom::Angle const & boresightRotAngle,
+        RotType const & rotType,
+        coord::Observatory const & observatory,
+        coord::Weather const & weather
+    ) :
+        _exposureId(exposureId),
+        _exposureTime(exposureTime),
+        _date(date),
+        _ut1(ut1),
+        _era(era),
+        _boresightRaDec(boresightRaDec),
+        _boresightAzAlt(boresightAzAlt),
+        _boresightAirmass(boresightAirmass),
+        _boresightRotAngle(boresightRotAngle),
+        _rotType(rotType),
+        _observatory(observatory),
+        _weather(weather)
+    {};
+
+    explicit VisitInfo(daf::base::PropertySet const & metadata);
```

Comment:
Thanks, @jonathansick. I made it a .rst file because I wanted the table support. I hope that is OK. Please feel free to rework the document to improve clarity; the current table is wider than I would like. Also, I could not figure out how to get rst to make low-key headers for the two tables and the footnotes so I used bold formatting instead (markdown wins there; pity its table support is so lame).

Also, the document is by no means complete in that it punts on describing the binary tables that may be present. I intended it to be complete in in describing LSST-specific header keywords (I did not attempt to describe the standard keywords such as SIMPLE, nor the standard WCS keywords).